src/battle/anim/water.c

```diff
@@ -549,3 +586,1941 @@ _080D3B38:\n\
+        sprite->data[0] = 6;
+        sprite->data[2] = (Random() & 31) + 65520 + sprite->pos1.x;
+        sprite->data[4] = (Random() & 31) + 65520 + sprite->pos1.y;
+        sprite->data[5] = ~(Random() & 7);
+        InitAnimArcTranslation(sprite);
+        sprite->callback = sub_80D4BA4;
+    }
+}
+
+void sub_80D4BA4(struct Sprite *sprite)
+{
+    if (TranslateAnimArc(sprite))
+    {
+        u16 i;
+        for (i = 0; i < 16; i++)
+        {
+            if (gTasks[i].func == sub_80D4988)
+            {
+                gTasks[i].data[10] = 1;
+                gTasks[i].data[8]--;
+                DestroySprite(sprite);
+            }
+        }
+    }
+}
+
+void sub_80D4BF0(struct Sprite *sprite)
+{
+    sprite->pos1.x = gBattleAnimArgs[0];
+    sprite->pos1.y = gBattleAnimArgs[1];
+    sprite->data[0] = gBattleAnimArgs[2];
+    sprite->data[1] = gBattleAnimArgs[3];
+    sprite->data[2] = gBattleAnimArgs[4];
+    sprite->data[3] = gBattleAnimArgs[5];
+    sprite->callback = sub_80D4C18;
+}
+
+void sub_80D4C18(struct Sprite *sprite)
+{
+    sprite->data[4] -= sprite->data[0];
+    sprite->pos2.y = sprite->data[4] / 10;
+    sprite->data[5] += sprite->data[1];
+    sprite->data[5] &= 0xff;
+    sprite->pos2.x = Sin(sprite->data[5], sprite->data[2]);
+    if (--sprite->data[3] == 0)
+    {
+        DestroyAnimSprite(sprite);
+    }
+}
+
+void sub_80D4C64(struct Sprite *sprite)
+{
+    sprite->data[3] += sprite->data[1];
+    sprite->data[4] += sprite->data[2];
+
+    sprite->pos2.x = sprite->data[3] >> 7;
+    sprite->pos2.y = sprite->data[4] >> 7;
+
+    if (--sprite->data[0] == 0)
+    {
+        FreeSpriteOamMatrix(sprite);
+        DestroySprite(sprite);
+    }
+}
+
+void sub_80D4CA4(struct Sprite *sprite)
+{
+    InitAnimSpritePos(sprite, 1);
+    sprite->data[1] = GetBattlerSpriteCoord(gAnimBankTarget, 2);
+    sprite->data[2] = GetBattlerSpriteCoord(gAnimBankTarget, 3);
+    sprite->data[3] = gBattleAnimArgs[2];
+    sprite->data[4] = gBattleAnimArgs[3];
+    sprite->callback = sub_80D4CEC;
+}
+
+void sub_80D4CEC(struct Sprite *sprite)
+{
+    int x = sprite->data[1] - sprite->pos1.x;
+    int y = sprite->data[2] - sprite->pos1.y;
+
+    sprite->pos2.x = sprite->data[0] * x / sprite->data[3];
+    sprite->pos2.y = sprite->data[0] * y / sprite->data[3];
+
+    if (++sprite->data[5] == sprite->data[4])
+    {
+        sprite->data[5] = 0;
+        sub_80D4D64(sprite, x, y);
+    }
+
+    if (sprite->data[3] == sprite->data[0])
+    {
+        DestroyAnimSprite(sprite);
+    }
+    sprite->data[0]++;
+}
+
+#ifdef NONMATCHING
+void sub_80D4D64(struct Sprite *sprite, int x, int y)
+{
+    s16 v1;
+    s16 sumX, sumY;
+    s16 rand1, rand2;
+    int i;
+
+    v1 = (sprite->data[0] + (sprite->data[0] >> 31)) << 15 >> 16; // sp
+    sumX = sprite->pos1.x + sprite->pos2.x; // r8
+    sumY = sprite->pos1.y + sprite->pos2.y; // r10
+
+    rand1 =  y + (Random() % 10) - 5; // r9
+    rand2 = -x + (Random() % 10) - 5; // r7
+
+    i = 0;
```

Comment:
are you sure all these loops aren't for ones?

Reply:
They may well be, I mostly opened this because I probably won't be around for a few weeks, if anyone can fix anything feel free